### PR TITLE
Migrate CI to GitHub workflow

### PR DESCRIPTION
**Pull request by tmichela**

### CODE REVIEW

File: .github/dependabot/constraints.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+euxfel_bunch_pattern
```
Comment (JamesWrigley) [inline .github/dependabot/constraints.txt:1]: I think these should also have version numbers, e.g. from `pip freeze`.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        packages: ${{ env.SYSTEM_PACKAGES }}
+
+    - uses: actions/cache@v3
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('**/setup.py') }}-${{ hashFiles('.github/dependabot/constraints.txt') }}
+
+    - name: Install Python dependencies
+      run: |
+        python3 -m pip install --upgrade pip
```
Comment (JamesWrigley) [inline .github/workflows/tests.yml:40]: Why is this is necessary? (not saying it should be removed, just curious)

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        python3 -m pip install .[test] --constraint .github/dependabot/constraints.txt
+
+    - name: Test with pytest
+      run: |
+        python3 -m pytest -v
+      env:
+        MPLBACKEND: agg
```
Comment (JamesWrigley) [inline .github/workflows/tests.yml:47]: Is this necessary? I don't think we needed it in the Gitlab CI.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    - name: Cache APT Packages
+      uses: awalsh128/cache-apt-pkgs-action@v1.3.0
+      with:
+        packages: ${{ env.SYSTEM_PACKAGES }}
+
+    - uses: actions/cache@v3
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('**/setup.py') }}-${{ hashFiles('.github/dependabot/constraints.txt') }}
```
Comment (JamesWrigley) [inline .github/workflows/tests.yml:36]: The first `hashFIles()` should be pointing to `pyproject.toml`, we don't have a `setup.py`.

**Review by JamesWrigley — CHANGES_REQUESTED** [review]
How annoying would it be to rollback the big rename? Asking because it'll probably be a bit painful rebasing the existing branches :sweat_smile: (yes, I'm aware this is entirely my fault)

**Comment by JamesWrigley** [comment]
Oh, and we can also get rid of `.gitlab-ci.yml` now.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (tmichela) [reply .github/workflows/tests.yml]: left over from copying the yaml file (was necessary in extra-data, but can't recall why)

**Comment by tmichela** [comment]
> How annoying would it be to rollback the big rename?

What do you suggest we do? Because I don't see the branches disapear anytime soon

**Review by JamesWrigley — APPROVED** [review]
I would suggest leaving the renaming till later if it's not too work much to revert. But I leave it to you, I don't feel very strongly about it. Other than that, LGTM!

**Comment by tmichela** [comment]
Clarified offline with James, leaving the renaming 📛 
Thanks for the review!

**Pull request merged**